src/ert/parsing/error_info.py

```diff
@@ -42,7 +44,7 @@ def set_context_keyword(
         self._attach_to_context(self._take(context, "keyword_token"))
         return self
 
-    def set_context_list(self, context_list: List[MaybeWithContext]) -> Self:
+    def set_context_list(self, context_list: Sequence[MaybeWithContext]) -> Self:
```

Comment:
Because `MaybeWithContext` is a union, we go with the immutable `Sequence` type here to enable variance (`List[FileContextToken]` is a subtype of `Sequence[MaybeWithContext]`, but not `List[MaybeWithContext]`.